flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -131,6 +137,11 @@
 
 	private final Resource resource;
 
+	/**
+	 * Executor for starting new yarn containers.
+	 */
+	private final Executor startContainerExecutor;
```

Comment:
This executor might need to be shutdown on 'onStop' is called.

Reply:
I would rather keep it the current way, for 2 reasons.
1. It does not cause any resource leak. The `ThreadPoolExecutor` is configured that pooled threads will be terminated if there's no tasks for 60s, and the executor itself is shutdown automatically if there is no remaining threads and it is no longer referenced.
2. The general `Executor` interface does not have a `shutdown` method. And I would like to keep the thread pool declared in the general type `Executor`, in case later we want to use a different executor implementation.